### PR TITLE
[TTAHUB-3402] Hook up QA widgets to BE

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This hooks up the new QA dashboard to the SSDI.

## How to test

- Review the code
- Make sure all widgets are displaying correctly
- Verify filters work correctly
- Sorting and exporting should work as it does on other pages

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3402
* https://jira.acf.gov/browse/TTAHUB-3457
* https://jira.acf.gov/browse/TTAHUB-3458
* https://jira.acf.gov/browse/TTAHUB-3459

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
Looked at this branch on dev, on the main QA dash landing page the "report ID" filter doesn't seem to do anything

**Comment by AdamAdHocTeam** [comment]
> The new dashboard routes 404 and I also don't see them in your code so I expect you have a bad merge w/main

Bad merge should now be fixed.

**Comment by thewatermethod** [comment]
> Verify filters work correctly

First two filters I tried on the landing page:
1) Report ID contains does not change the result set as far as I can tell, at least on the overview widget
2) Created on throws an error
Screenshot 2024-10-17 at 9 27 49 AM

File: frontend/src/fetchers/ssdi.js
Comment (thewatermethod) [inline frontend/src/fetchers/ssdi.js:49]: why do we need both?

File: frontend/src/fetchers/ssdi.js
Comment (thewatermethod) [inline frontend/src/fetchers/ssdi.js:81]: do we need to account for this case (dataSetSelection is falsy?) If so, did we add a test.
Note also there is no need to check for length: if dataSetSelection is empty, the map will be an empty array and will join out to be the same as your else statement

File: frontend/src/hooks/useWidgetSorting.js
Comment (thewatermethod) [inline frontend/src/hooks/useWidgetSorting.js:31]: Can you leave a comment explaining why we need to optionally override the sortConfig direction?

Comment (thewatermethod) [reply frontend/src/hooks/useWidgetSorting.js:31]: Just seems like you could mutate the object that already has the direction in it instead of adding another parameter and all this additional logic

File: frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js
Comment (thewatermethod) [inline frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js:110]: No test for this case I don't think

File: frontend/src/widgets/PercentageActivityReportByRole.js
Comment (thewatermethod) [inline frontend/src/widgets/PercentageActivityReportByRole.js:27]: I left the loading attribute off these graphs on purpose; we don't really want all those announcements/mini "loading" spinners since the whole page is wrapped in a loader. There's an outstanding accessibility ticket to fix it. everywhere.

File: frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js
Comment (thewatermethod) [inline frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js:179]: can we fix the spelling of this method?

File: frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js
Comment (thewatermethod) [inline frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js:173]: no test for this case

File: frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js
Comment (thewatermethod) [inline frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js:124]: same comments as before, no test for this case that I can see, and the function has a typo

**Comment by AdamAdHocTeam** [comment]
> > Verify filters work correctly
> 
> First two filters I tried on the landing page:
> 
> 1. Report ID contains does not change the result set as far as I can tell, at least on the overview widget
> 2. Created on throws an error
>    Screenshot 2024-10-17 at 9 27 49 AM

This should be fixed in Garrett's latest push.

File: frontend/src/fetchers/ssdi.js
Comment (AdamAdHocTeam) [reply frontend/src/fetchers/ssdi.js:49]: I was waiting till Garrett updated the BE, I have removed it.

File: frontend/src/fetchers/ssdi.js
Comment (AdamAdHocTeam) [reply frontend/src/fetchers/ssdi.js:81]: Removed checks.

File: frontend/src/hooks/useWidgetSorting.js
Comment (AdamAdHocTeam) [reply frontend/src/hooks/useWidgetSorting.js:31]: I added a comment.

This seems to just take the opposite of whats currently set in the sortConfig (arrow sort).

**Comment by thewatermethod** [comment]
FYI @AdamAdHocTeam, @GarrettEHill Dashboard no longer loads any data at all

File: frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js
Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js:110]: added

File: frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js
Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js:179]: yes

File: frontend/src/widgets/PercentageActivityReportByRole.js
Comment (AdamAdHocTeam) [reply frontend/src/widgets/PercentageActivityReportByRole.js:27]: Ok I removed loaders from all of the individual widgets and just added one loader on the dashboard itself.

File: frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js
Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/RecipientsWithClassScoresAndGoals/index.js:173]: added

File: frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js
Comment (AdamAdHocTeam) [reply frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js:124]: added test fixed typo

**Comment by AdamAdHocTeam** [comment]
> FYI @AdamAdHocTeam, @GarrettEHill Dashboard no longer loads any data at all

@thewatermethod this should be good now @GarrettEHill  resolved the BE issues.

File: frontend/src/widgets/PercentageActivityReportByRole.js
Comment (AdamAdHocTeam) [reply frontend/src/widgets/PercentageActivityReportByRole.js]: This has been cleaned up.

**Comment by thewatermethod** [comment]
I noticed that the total number of reports (11,510) on delivery method and percentage of activity reports by role does not change, I think that needs to be hooked up to the backend.

## Filters that error:
- Created on
- Date started
- Date ended
- Group
- Reasons
- Recipient name
- State, if you pick more than one

## Filters that do not seem to update the data (none of the charts, even the ones that would make sense to update, lke the root cause chart)

Please double check these are being applied

- Goal status
- FEI root cause
- Goal text
- Target population
- Number of recipients
- Specialist roles
- TTA type

These only apply to the landing page; when I started picking through the sub pages I found similar issues but I'm not going to document them rn

**Comment by AdamAdHocTeam** [comment]
> @AdamAdHocTeam @GarrettEHill I noticed that the total number of reports (11,510) on delivery method and percentage of activity reports by role does not change, I think that needs to be hooked up to the backend.
> 
> ## Filters that error:
> * Created on
> * Date started
> * Date ended
> * Group
> * Reasons
> * Recipient name
> * State, if you pick more than one
> 
> ## Filters that do not seem to update the data (none of the charts, even the ones that would make sense to update, lke the root cause chart)
> Please double check these are being applied
> 
> * Goal status
> * FEI root cause
> * Goal text
> * Target population
> * Number of recipients
> * Specialist roles
> * TTA type
> 
> These only apply to the landing page; when I started picking through the sub pages I found similar issues but I'm not going to document them rn

@GarrettEHill can you have a look at this on Monday seems we are missing data points and filters.

File: frontend/src/widgets/RecipientsWithClassScoresAndGoalsWidget.js
Comment (thewatermethod) [inline frontend/src/widgets/RecipientsWithClassScoresAndGoalsWidget.js:91]: spelling

File: frontend/src/components/FiltersNotApplicable.scss
Comment (thewatermethod) [inline frontend/src/components/FiltersNotApplicable.scss:18]: Is this the right place for this, since it will apply to all tooltips? Nothing here is specific to FiltersNotApplicable

File: frontend/src/widgets/OverviewWidgetField.js
Comment (thewatermethod) [inline frontend/src/widgets/OverviewWidgetField.js:36]: Nice to see this turned into a component

File: frontend/src/widgets/HorizontalTableWidget.scss
Comment (thewatermethod) [inline frontend/src/widgets/HorizontalTableWidget.scss:135]: at the design/dev meeting they said they didn't want this, did they change their mind?

File: frontend/src/widgets/HorizontalTableWidget.scss
Comment (thewatermethod) [inline frontend/src/widgets/HorizontalTableWidget.scss:122]: Can we use a class to select these instead of :first-child, nth-child?

File: frontend/src/components/FiltersNotApplicable.scss
Comment (AdamAdHocTeam) [reply frontend/src/components/FiltersNotApplicable.scss:18]: I added a class for the component.

File: frontend/src/widgets/HorizontalTableWidget.js
Comment (AdamAdHocTeam) [reply frontend/src/widgets/HorizontalTableWidget.js]: updated target to self.

File: frontend/src/widgets/HorizontalTableWidget.scss
Comment (AdamAdHocTeam) [reply frontend/src/widgets/HorizontalTableWidget.scss:135]: They don't want a border on the first column?

File: frontend/src/widgets/HorizontalTableWidget.scss
Comment (AdamAdHocTeam) [reply frontend/src/widgets/HorizontalTableWidget.scss:122]: Its a map will leave as is for now.

Comment (AdamAdHocTeam) [reply frontend/src/widgets/HorizontalTableWidget.scss:128]: I put this back so the border only shows in larger views.

**Comment by thewatermethod** [comment]
Screenshot 2024-10-25 at 2 47 22 PM

**Comment by thewatermethod** [comment]
Screenshot 2024-10-25 at 2 55 16 PM

**Comment by thewatermethod** [comment]
Screenshot 2024-10-25 at 2 57 45 PM

**Comment by thewatermethod** [comment]
Screenshot 2024-10-25 at 2 58 22 PM

**Comment by thewatermethod** [comment]
Screenshot 2024-10-25 at 2 59 21 PM

**Comment by thewatermethod** [comment]
Everything is looking really good, we just need to make sure all the filters work. Let me know if you want an approval on this PR and to defer that work to a separate one. (Obviously we won't go live until everything is hooked up anyways)

**Comment by AdamAdHocTeam** [comment]
> Everything is looking really good, we just need to make sure all the filters work. Let me know if you want an approval on this PR and to defer that work to a separate one. (Obviously we won't go live until everything is hooked up anyways)

@thewatermethod for Monday, I have fixed all mentioned filters.

**Review by thewatermethod — APPROVED** [review]
I can't access the CLASS page, please manually check all filters on that page before merging.

Everything looks great

**Pull request merged**
